### PR TITLE
Fix #3932: ANSI color leak in repl

### LINKED ISSUE #3932: REPL pretty-printed output for implicit function types leak ANSI color codes.
```scala 
scala> def fun[T](x: T): implicit List[T] => Int = ???
def fun[T](x: T): 33mimplicit 0mList[T] => Int
```

### PR DESCRIPTION
**Pull request by OlivierBlanvillain**

### CODE REVIEW

**Comment by allanrenucci** [comment]
I believe the REPL had its syntax highlighting mechanism before we introduced colors in Printer.

I think a better fix is to make the REPL only relies on the mechanism from the Printers (i.e. replace call to `SyntaxHighlighting` in ReplDriver by call to `show`).

**Comment by OlivierBlanvillain** [comment]
Unless we want to stop highlighting code "as it's been written", we won't be able to get ride of the REPL syntax highlighting.

Given than we have two syntax highlighters it's not clear to me what we would gain by using the one from RefinedPrinter over the REPL one.

**Comment by allanrenucci** [comment]
The syntax highlighter from the printers is supposed to be more fined grained (it knows the semantic of what it is printing) and is also configurable (using the compiler options).

The issue in #3932 is because we are currently calling `SyntaxHighlighting.apply` on the result of a show and it is coloring twice. We call show anyway, so why not let it do syntax highlighting.

You are right for code  "as it's been written", we have to keep using `SyntaxHighlighting.apply` because in this case we consume a `String` and not  a `Tree` or a `Symbol`. However this is not done in the `ReplDriver` but in the `AmonniteReader`. I didn't mean to change that

**Comment by smarter** [comment]
> You are right for code "as it's been written", we have to keep using SyntaxHighlighting.apply because in this case we consume a String and not a Tree or a Symbol.

Side-note: we could use the regular Parser to handle this String. Our parser is supposed to recover from errors (needed for the IDE to work well), so it could also work for syntax highlighting as you type in the REPL.

File: compiler/test-resources/repl/erased [hunk 1]
```diff
@@ -0,0 +1,2 @@
+scala> def f(erased a: Int): Int = ???
+def f(erased a: Int): Int
```
[the remark this answers is not in the document]
Comment (OlivierBlanvillain) [reply compiler/test-resources/repl/erased]: Done!

**Comment by nicolasstucki** [comment]
@OlivierBlanvillain is this still a WIP?

**Comment by OlivierBlanvillain** [comment]
It's good to go!

**Pull request merged**